src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/AuthenticationHelper.cs

```diff
@@ -251,11 +252,19 @@ await TrySetDigestAuthToken(request, challenge.Credential, digestResponse, isPro
                                     {
                                         try
                                         {
+                                            if (NetEventSource.IsEnabled)
+                                            {
+                                                NetEventSource.Info(pool.PreAuthCredentials, $"Adding Basic credential to cache, uri={authUri}, username={challenge.Credential.UserName}");
```

Comment:
Is challenge.Credential guaranteed to be non-null here?

Reply:
Yes, as long as the TryGetAuthentication*() method returns true. And the logging is only done when that method returns true.